### PR TITLE
runtime: Add tests for Pic

### PR DESCRIPTION
**Pull request by ArthurHeymans**
This adds 2 tests:
- Check that a wrongly configured ext int vector table results in an NMI
- Properly configured external interrupts wake from halt

Follow-up patches will use this to get accurate sw-emulation right.

### CODE REVIEW

File: runtime/test-fw/src/pic_tests.rs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+#![no_main]
+#![no_std]
+
+use caliptra_drivers::{cprintln, Pic, SocIfc};
+use caliptra_registers::{el2_pic_ctrl::El2PicCtrl, soc_ifc::SocIfcReg};
+use caliptra_test_harness::test_suite;
+
+#[cfg(target_arch = "riscv32")]
+core::arch::global_asm!(include_str!("ext_intr.S"));
```
Comment (korran) [inline runtime/test-fw/src/pic_tests.rs:11]: Can you get away with include_str!("../../extr_intr.S") and delete this file? I don't like copy-and-paste.

Comment (ArthurHeymans) [reply runtime/test-fw/src/pic_tests.rs:11]: Ah sure. Older iterations had similar code to NMI to vector into rust eventually. That didn't work so I had to revert it. With the new insights from the meeting I might give that another try.

Comment (ArthurHeymans) [reply runtime/test-fw/src/pic_tests.rs:11]: I added it back since it works with a few nop cycles after enabling global interrupts.

File: runtime/test-fw/src/ext_intr.S [hunk 2]
```diff
@@ -0,0 +1,171 @@
+
+.section .init.text, "ax"
+.align 2
+_ext_intr_handler:
```
Comment (korran) [inline runtime/test-fw/src/ext_intr.S:4]: I don't understand what we're testing here. What does this have to do with runtime firmware? It seems like this is more to do with testing the hardware model itself...

Comment (korran) [reply runtime/test-fw/src/ext_intr.S:4]: Typically those tests go into hw-model/.

Comment (ArthurHeymans) [reply runtime/test-fw/src/ext_intr.S:4]: Thanks that makes sense. I'll move it to hw-model. The goal is indeed to make the emulator more accurately match hardware.

Comment (ArthurHeymans) [reply runtime/test-fw/src/ext_intr.S:4]: The runtime test firmware setup is more convenient than the test-harness rom as things like nmi is already set up. I can make the tests include another .S file that does all that ?

**Comment by ArthurHeymans** [comment]
The sw-emulator patch for PIC pas this test btw.
